docs/website/docs/developers/performance/profiling-with-tracy.md

```diff
@@ -4,72 +4,327 @@ icon: material/chart-line
 
-[Tracy](https://github.com/wolfpld/tracy) is a profiler that puts together in a
-single view both instrumentation and system profiling (sampling, systrace). It's
-key to understand the nuance here.
+## :octicons-book-16: Overview
 
-* *Instrumentation* is code built into the process being profiled, collecting
-  timestamps at the start and end of "zones". Once it's enabled at build time,
-  it typically just works &mdash; it is a part of our application logic just
-  like anything else, so there's no reason why it would not work.
-* *Sampling* and *SysTrace* rely on specific
-  system features to collect information on what is *actually* running. These
-  rely on OS and binary (ELF) file features, so they can take a bit more care to
-  get to work properly.
+[Tracy](https://github.com/wolfpld/tracy) is a hybrid instrumentation and
+sampling profiler that IREE uses for performance analysis.
 
-There are two components to Tracy. They communicate over a TCP socket.
+[![Tracy overview](https://github.com/openxla/iree/assets/4010439/3df4f682-9215-4f55-b7b0-5fe2dce860b0)](https://github.com/openxla/iree/assets/4010439/3df4f682-9215-4f55-b7b0-5fe2dce860b0)
+
+* *Instrumentation* is generic code built into the program being profiled,
+    recording zone start and end timestamps where a developer requests them:
+
+    [![Tracy instrumentation](https://github.com/openxla/iree/assets/4010439/70706e6d-d253-4ca3-85bf-165f23d24635)](https://github.com/openxla/iree/assets/4010439/70706e6d-d253-4ca3-85bf-165f23d24635)
+
+* *Sampling* collects program state and information about the machine using
+    platform-specific APIs at a regular sampling frequency. Sampled data
+    includes callstacks, hardware counters, and more:
+
+    [![Tracy sampling](https://github.com/openxla/iree/assets/4010439/d637a692-8bfa-42c4-bd0d-75ea3dce2e69)](https://github.com/openxla/iree/assets/4010439/d637a692-8bfa-42c4-bd0d-75ea3dce2e69)
+
+Tracy uses a client-server model with communication over a TCP socket:
 
 * The "client" is the program being profiled.
-* The "server" is:
-    * Either the Tracy profiler UI (which we build as `iree-tracy-profiler`),
-    * Or the Tracy command-line capture tool (`iree-tracy-capture`) that can
-        save a trace for later loading in the Tracy profiler UI.
-
-## The Tracy manual
-
-The primary source of Tracy documentation, including for build instructions, is
-a PDF manual that's part of each numbered release.
-[Download](https://github.com/wolfpld/tracy/releases/latest/download/tracy.pdf)
-or
-[view in browser](https://docs.google.com/viewer?url=https://github.com/wolfpld/tracy/releases/latest/download/tracy.pdf).
-
-## Overview
-
-We will go through each steps below, but here is an overview. It highlights the
-simpler subset of instructions when only instrumentation is needed, vs. the
-additional steps needed when Sampling is also wanted.
-
-Component | Instrumentation only | Instrumentation and Sampling
---- | --- | ---
```

Comment:
> The big table at the start of the old doc served those two purposes (make it easy to check that one had not missed a step for sampling, and convey the simpler instructions for Instrumentation vs heavier instructions for Sampling).

I used to like the large table, but I think it's grown too big. For someone getting started, it's a large information dump that doesn't make it clear if the options are independent from one another.

Specifically, I wanted to:
* Remove the "Build Tracy capture" and "Build Tracy profiler" rows and lean on the upstream build + documentation there. If we're not quite ready for that due to capstone and other Linux build woes, then I'll settle for an appendix section in our docs.
* Simplify "Build the IREE compiler" and "Build IREE device binaries" - All people should really need is a standard source build with `-DIREE_ENABLE_RUNTIME_TRACING` or the Python bindings from pip.

I think the new "capturing a trace" and "quickstart" sections https://scotttodd.github.io/iree/developers/performance/profiling-with-tracy/#capturing-a-trace
cover most of the setup and then people can branch from there.

I'll try reordering sections a bit and will make sure each usage style is explained well enough.

Reply:
@bjacob I made a few changes. Curious to hear your thoughts.

![image](https://github.com/openxla/iree/assets/4010439/055aa01d-bafb-46c2-8f62-be6c76e55200)

Mainly I moved the "Including more information" tips from a section below to inlined example settings with annotations. I also highlighted some key lines in each code block.

Reply:
I guess we could also add a more explicit section on instrumentation vs sampling so people don't go down the full sampling path if they aren't particularly looking to use that data.

Reply:
Thanks! I found the little `+` icons (neat -- I didn't know how to do that) and when clicking them I did see the tip pop up, which is good... but leaves me wondering if I would have discovered that while reading it.  Most likely, my lazy overworked tutorial-reading self would have copied and pasted commands into a terminal without noticing the `+` icon.

Reply:
> but leaves me wondering if I would have discovered that while reading it

I was hoping the highlights would help draw attention there, but yeah... I share this concern. Thinking I might restructure a bit more. Could introduce some repetition in content (dedicated section _and_ `+` icon annotations)

Reply:
Added more information upfront about what sampling requires.

Reply:
Removed the `+` icons in favor of code block comments. Should be easier to copy/paste now: 
![image](https://github.com/openxla/iree/assets/4010439/6ede64ac-434d-4e6e-8e60-320fd8220cd9)